### ISSUE TITLE
Image scan feature

### ISSUE THREAD

**Issue by StrongMonkey** [rancher/fleet#378]
We need to add image scan feature into fleet. Allowing user to scan an image respository and write back the image changes to git repo. This allows user to automatically apply the manifest with new image tag once image is released.

See similar feature: https://fluxcd.io/docs/guides/image-update/

**Comment by StrongMonkey** [comment]
@izaac You can test this featue following the doc here https://github.com/rancher/fleet/pull/413

**Comment by SheilaghM** [comment]
@PennyScissors - please validate this feature for QA.  Daishan left instructions on testing, see below.  He's also back Monday if you have questions.

**Comment by PennyScissors** [comment]
Tested in rancher/rancher:master-head 9e0d1b2

Steps:
1. Create k3s cluster (all default values).
2. Add a Docker Hub secret to the `default` namespace in the k3s cluster for the fleet image scan policy configuration.
3. Create a Git Repo ([what I used](https://github.com/PennyScissors/testing/tree/master/fleet/image-scan)) in the cluster's Continuous Integration and change the target to the k3s cluster (not all clusters in workspace).
4. Verify the deployment gets deployed successfully
5. Bump the image tag and push to Docker Hub.
6. Verify the deployment gets updated to match the new image tag.

Problems:
1. After step 3, I see the following error, but is not consistent. As a workaround, forcing an update on the Git Repo seems to fix it.
```
Time="2021-07-23T17:03:31Z" level=info msg="updated: fleet-default/testing-fleet-image-scan\n" time="2021-07-23T17:03:31Z" level=info msg="updated (scan): fleet-default/imagescan-testing-fleet-image-scan-0\n" time="2021-07-23T17:03:31Z" level=info msg="created (scan): fleet-default/testing-fleet-image-scan\n" time="2021-07-23T17:03:31Z" level=fatal msg="resource name may not be empty" 
```
2. Pushed a new tag for the hello-world image and the change was not detected.
Deployment still using scresporancher/hello-world:1.0.0
![image](https://user-images.githubusercontent.com/8951271/126819730-fc07eed0-0bfa-40a1-9f39-ef557e9fd488.png)
Docker Hub with a new scresporancher/hello-world:1.0.1 image available
![image](https://user-images.githubusercontent.com/8951271/126820104-3f8f07c3-1b02-4612-920b-3791bdd6bd6e.png)

**Comment by StrongMonkey** [comment]
@PennyScissors This should be fixed in v0.3.6-rc7

**Comment by PennyScissors** [comment]
Validated following the same steps described above. Changes were detected and the deployment manifest, and workload was updated accordingly. The previous error related to `resource name may not be empty` was not present.

**BEFORE** bumping image tags and pushing:

![Screen Shot 2021-07-29 at 2 53 19 PM](https://user-images.githubusercontent.com/8951271/127573442-6c09ee86-d862-4c11-a81a-6e055e2dbe32.png)

![image](https://user-images.githubusercontent.com/8951271/127570694-2d7532cb-0044-424f-9e6f-c6d837457b7a.png)


**AFTER** bumping image tags and pushing:

![image](https://user-images.githubusercontent.com/8951271/127573318-cca08b16-6b0c-45d5-bf60-49795786e96f.png)

![image](https://user-images.githubusercontent.com/8951271/127573276-5d5b9727-da7c-4349-ab04-6ffe23e33842.png)
